compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -466,10 +476,23 @@ pub fn make_format_args(
             } else {
                 if let Some(expr) = args.explicit_args()[i].expr.to_ty() {
                     if let Some(symbol) = expr.kind.is_simple_path() {
-                        let current_arg = symbol.as_str().to_owned();
-                        let current_arg_ph = format!("{{{current_arg}}}");
-                        if current_arg.len() > 0 && fmt_str.contains(current_arg_ph.as_str()) {
-                            duplicate_explicit_arg.push(args.explicit_args()[i].expr.span);
+                        for piece in &pieces {
+                            if let NextArgument(arg) = piece {
+                                if let ArgumentNamed(specifier) = arg.position {
+                                    if specifier == symbol.to_string() {
+                                        redundant_expl_args.push({
+                                            RedundantArgDiagBuilder {
+                                                arg_span: expr.span,
+                                                fmt_span: InnerSpan {
+                                                    start: arg.position_span.start,
+                                                    end: arg.position_span.end,
+                                                },
+                                                fmt_str: specifier,
+                                            }
+                                        });
+                                    }
+                                }
+                            }
                         }
                     }
                 }
```

Comment:
We can take the chance to leverage let chains and reduce the indentation levels of this code:
```rust
                if let Some(expr) = args.explicit_args()[i].expr.to_ty()
                    && let Some(symbol) = expr.kind.is_simple_path()
                {
                    let current_arg = symbol.as_str().to_owned();
                    for piece in &pieces {
                        if let NextArgument(arg) = piece
                            && let ArgumentNamed(specifier) = arg.position
                        {
                            if specifier == symbol.to_string() {
                                redundant_expl_args.push(RedundantArgDiagBuilder {
                                    arg_span: expr.span,
                                    fmt_span: InnerSpan {
                                        start: arg.position_span.start,
                                        end: arg.position_span.end,
                                    },
                                    fmt_str: specifier,
                                });
                            }
                        }
                    }
                }
```